### DISCUSSION TITLE
effect of div(phi,U) scheme on results verity

### DISCUSSION THREAD

**Discussion by Ali19881366** [fppimenta/rheoTool#64] [General]
Hello everybody

I solved the problem of flow in a curved duct and captured the secondary flow patterns and I could see the deviation in the velocity profile using simpleFoam as reported by Bara in his thesis (*). However, when I used rheoFoam the velocity profile became symmetric (non-physical results). I checked many things, and I found that is because of the div(phi,U) scheme that was set on :GaussDefCmpw none. I changed it to GaussDefCmpw cubista and got the correct results. This scheme (GaussDefCmpw none) is used in many of the viscoelastic cases. Why does this scheme have such a effect on the results? why doesn't affect the accuracy of the results of viscoelastic cases?

Thanks

(*) An experimental and numerical study of the Dean problem: flow development towards two-dimensional multiple solutions

**Comment by fppimenta** [comment]
Not an issue.

Have you ever read the user-guide or at least the section related with class GaussDefCmpw? Never use anything as a blackbox.

**Reply by Ali19881366** [reply]
Of course, using a numerical solver as a black box is not a good way!! But, I have been working with the previous versions of rheoTool, and recently I installed the new version. Also, I read the user guide of those versions, and I didn't know about this new option. As I remember, there is no "none" option in schemes of the first version of rheoTool, not to discretize the convection terms in creeping flow regimes. So, I didn't check the schemes. That's it!

**Comment by fppimenta** [comment]
This is the link to very first version (commit) of GaussDefCmpw, released 5 years ago: 

https://github.com/fppimenta/rheoTool/blob/0afe19e5c2812a33549566e0185b6806d4facab0/of222/src/libs/gaussDefCmpwConvectionScheme/gaussDefCmpwConvectionScheme.C

As you can see, the option was already there and has been there always in all releases and versions. The same goes to the user-guide.

**Reply by Ali19881366** [reply]
Ok. I was wrong. Although I even read your papers about the formulation of the log-conformation approach and the CUBISTA scheme (and I did not use the rheoTool as a black box), It seems that I didn't read this part carefully.
